cli/kata-check_amd64.go

```diff
@@ -5,13 +5,43 @@
 
 package main
 
+/*
+#include <stdint.h>
+
+// Current Linux mainstream kernel doesn't have support for ACRN.
+// Due to this several macros are not defined in Linux headers.
+// Until the support is available, directly use the value instead
+// of macros.
+//https://github.com/kata-containers/runtime/issues/1784
+const int ioctl_ACRN_CREATE_VM  = 0x43000010;
+
+struct acrn_create_vm {
+	uint16_t vmid;
+	uint16_t reserved0;
+	uint16_t vcpu_num;
+	uint16_t reserved1;
+	uint8_t	 uuid[16];
+	uint64_t vm_flag;
+	uint64_t req_buf;
+	uint8_t  reserved2[16];
+}__attribute__((aligned(8)));
+
+struct acrn_create_vm create_vm = {
+	.uuid = {0xd2U, 0x79U, 0x54U, 0x38U, 0x25U, 0xd6U, 0x11U, 0xe8U,	\
+		0x86U, 0x4eU, 0xcbU, 0x7aU, 0x18U, 0xb3U, 0x46U, 0x43U}
+};
+*/
+import "C"
```

Comment:
Why cgo? All you need are the two data structures and the ioctl number. You can easily define them in go.

Reply:
Hi @bergwolf, 1) without cgo is it still possible to ensure that the address of the struct will be 8-byte  memory aligned? Not sure what is the replacement for _attribute_(aligned(8)). 2) I only need to initialize UUID to check if a VM can be created using ACRN but declaring  acrn_create_vm as go struct results in make-check failing with unused parameter error for other variables in the struct. 3) Would like to retain the same name of the member variables as defined in the kernel, but declaring the struct as go, requires we remove "_" from member names.

Reply:
@vijaydhanraj 
1) The [go specification](https://golang.org/ref/spec#Size_and_alignment_guarantees) requires that 
> 2. For a variable x of struct type: unsafe.Alignof(x) is the largest of all the values unsafe.Alignof(x.f) for each field f of x, but at least 1.

So the acrn_create_vm struct is 8-byte aligned on 64 bit architectures as it has `uint64_t` in it.

2). you can use `//nolint` to disable it, see https://github.com/golangci/golangci-lint#nolint

Reply:
> without cgo is it still possible to ensure that the address of the struct will be 8-byte memory aligned? Not sure what is the replacement for attribute(aligned(8)). 

It depends on where and how the struct variable is defined.  From [the go docs](https://golang.org/pkg/sync/atomic/#pkg-note-BUG) 

> The first word in a variable or in an allocated struct, array, or slice can be relied upon to be 64-bit aligned.

Apart from that there's currently no "safe" way to request alignment for a variable.

Reply:
Thanks @bergwolf , @markdryan. Have removed the cgo and created go structs. Have also used nolint to suppress few variable names.